src/Libraries/PythonNodeModels/PythonEvaluationHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+
+namespace PythonNodeModels
+{
+    /// <summary>
+    /// Enum of possible values of python engine versions.
+    /// TODO: Remove when dynamic loading logic is there then we no longer need a hard copy of the options
+    /// </summary>
+    public enum PythonEngineVersion
+    {
+        IronPython2,
+        CPython3
+    }
+
+    /// <summary>
+    /// Singleton class that other class can access and use for query loaded Python Engine info.
+    /// TODO: Make the Singleton class public when Dynamic loading is also ready.
+    /// </summary>
+    internal sealed class PythonEvaluationHelper
+    {
+        /// <summary>
+        /// Use Lazy<![CDATA[PythonEvaluationHelper]]> to make sure the Singleton class is only initialized once
+        /// </summary>
+        private static readonly Lazy<PythonEvaluationHelper>
+            lazy =
+            new Lazy<PythonEvaluationHelper>
+            (() => new PythonEvaluationHelper());
+
+        /// <summary>
+        /// The actual instance stored in the Singleton class
+        /// </summary>
+        internal static PythonEvaluationHelper Instance { get { return lazy.Value; } }
+
+        // TODO: The following fields might be removed after dynamic loading applied
+        internal static bool IsIronPythonEnabled = false;
+        internal static string IronPythonEvaluatorClass = "IronPythonEvaluator";
+        internal static string IronPythonEvaluationMethod = "EvaluateIronPythonScript";
+
+        internal static bool IsCPythonEnabled = false;
+        internal static string CPythonEvaluatorClass = "CPythonEvaluator";
+        internal static string CPythonEvaluationMethod = "EvaluatePythonScript";
+
+        /// <summary>
+        /// Singleton class initialization logic which will be run in a lazy way the first time Dynamo try to evaluate a Python node
+        /// </summary>
+        private PythonEvaluationHelper()
+        {
+            var assems = AppDomain.CurrentDomain.GetAssemblies();
+            // Currently we are using try-catch to validate loaded assembly and evaluation method exist
+            // but we can optimize by checking all loaded types against evaluators interface later
+            try
+            {
+                var IronPythonAssem = assems.First(x => x.FullName.Contains("DSIronPython"));
```

Comment:
I was thinking to check if the IronPython extension is loaded as a way to tell if DSIronPython is loaded or not. My concern with this approach is that there are ways that DSIronPython can be loaded in the appdomain but this does not guarantee that it is imported into the VM. The only way it will be imported into the VM right now, is in the presence of the extension.

Reply:
Also, it being imported in the VM implies that it will also be loaded in the appdomain.

Reply:
hmm I thought about it briefly but I ended up this way because even extension is there, if DSIronPython is not loaded then IronPython is still not enabled. I think either of the approach here is ideal but should be replaced with the Dynamic loading logic based on interface type check later. What do you think?

Reply:
@QilongTang you got a point too.

Reply:
I just used the following code from an extension to determine whether the package was loaded or not:
```
var installedPackages = pmExtension.PackageLoader.LocalPackages;
var isPackageLoaded = installedPackages.Any(x =>
                x.Name == "IronPythonPackageName" && x.VersionName == "packageVersion");
```
where `pmExtension` is a reference to the package manager extension object.

Reply:
@aparajit-pratap I think that should serve your need. Once this PR is merged, you should be able to call `PythonEngineSelector.Instance.IsIronPythonEnabled`  from extension to also do the check.

Reply:
@QilongTang I actually prefer to do it this way as I think it's cleaner and was proposing the same solution to you.

Reply:
@aparajit-pratap hmm understood, the reason I am doing it this way is because
1. We may not have access to package manager in a lot of the places where we need to call the API
2. We need to validate the evaluation method exist so that Dynamo do not get tricked by some dll just with the same name
3. These logic will be replaced with Dynamic loading later, so in the future we will be checking against in memory types implementing a particular interface but not just the method exist. That would be the most accurate in my opinion.
4. The IronPython extension only exist for Dynamo 2.7 and beyond, checking extension exist may not apply to lower version, although this depends on if there will be extension author implementing similar APIs.

To conclude, I believe the API will be improved in the future for a more generic and dynamic way of determining the loaded Python Engine.